ios/Classes/DebugNotificationHelper.swift

```diff
@@ -0,0 +1,86 @@
+        let message =
+        """
+        Starting Dart/Flutter with following params:
+         • callbackHandle: '\(callBackHandle)'
+         • callBackName: '\(callbackInfo.callbackName ?? "not found")'
+         • callbackClassName: '\(callbackInfo.callbackClassName ?? "not found")'
+         • callbackLibraryPath: '\(callbackInfo.callbackLibraryPath ?? "not found")'
+        """
+        DebugNotificationHelper.scheduleNotification(identifier: identifier.uuidString,
+                                                     title: startDate.formatted(),
+                                                     body: message,
+                                                     icon: .startWork)
+    }
+    
+    static func showCompletedFetchNotification(identifier: UUID, completedDate: Date, result: UIBackgroundFetchResult, elapsedTime: TimeInterval) {
+        let message =
+        """
+        Perform fetch completed:
+         • Elapsed time: \(elapsedTime.formatted())
+         • Result: UIBackgroundFetchResult.\(result)
+        """
+        DebugNotificationHelper.scheduleNotification(identifier: identifier.uuidString,
+                                                     title: completedDate.formatted(),
+                                                     body: message,
+                                                     icon: result == .newData ? .success : .failure)
+    }
+    
+    
+    // MARK: -  Private helper functions
+    
+    private static func scheduleNotification(identifier: String, title: String, body: String, icon: ThumbnailGenerator.ThumbnailIcon) {
+        guard UserDefaultsHelper.getIsDebug(), #available(iOS 10.0, *) else {
+            logInfo("\(logPrefix) \(#function): plugin is not running in debug mode or on iOS 9 or lower")
+            return
+        }
+
+        DebugNotificationHelper.requestAuthorization()
+        let notification = UNMutableNotificationContent()
+        notification.title = title
+        notification.body = body
+        notification.threadIdentifier = SwiftWorkmanagerPlugin.identifier
+        if let thumbnail = ThumbnailGenerator.createThumbnail(with: icon) {
+            notification.attachments = [thumbnail]
+        }
+        let immediateFutureTrigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
+        let notificationRequest = UNNotificationRequest(identifier: identifier, content: notification, trigger: immediateFutureTrigger)
+        
+        UNUserNotificationCenter.current().add(notificationRequest, withCompletionHandler: nil)
+    }
+    
+    private static func requestAuthorization() {
+        guard UserDefaultsHelper.getIsDebug(), #available(iOS 10.0, *) else {
+            logInfo("\(logPrefix) \(#function): plugin is not running in debug mode or on iOS 9 or lower")
+            return
+        }
+        
+        UNUserNotificationCenter.current().requestAuthorization(options: [.sound, .alert]) { (success, error) in
```

Comment:
From the [docs](https://developer.apple.com/documentation/usernotifications/unusernotificationcenter/1649527-requestauthorization):

> Always call this method before scheduling any local notifications and before registering with the Apple Push Notification service.

So perhaps the scheduling of the local notification should be done in the completion and  if` success == true`?